### ISSUE TITLE
False negative about SimplifyBooleanReturn

### ISSUE THREAD

**Issue by Zustin** [checkstyle/checkstyle#11580]
I have read check documentation: https://checkstyle.sourceforge.io/config_coding.html#SimplifyBooleanReturn
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac Test.java

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <property name="severity" value="error"/>
  <property name="fileExtensions" value="java, properties, xml"/>
  <module name="TreeWalker">
    <module name="SimplifyBooleanReturn"/>
  </module>
</module>
        
/var/tmp $ cat Test.java
public class Test {
  private boolean cond;
  public boolean check1() {
    if (cond) { // should report a warning here
      return true || false;
    } else {
      return false;
    }
  }
}

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.1-all.jar -c config.xml Test.java
Starting audit...
Audit done.
```

**Describe what you expect in detail.**
Hi, I found a false negative about the rule SimplifyBooleanReturn. For the code example above, CheckStyle should have report a warning in line 5 because this is a literal expression and it is directly equal to `true` in bytecode level. Hence, I think this is a false negative. Thanks for your consideration.

**Comment by romani** [comment]
Can you share real code where it can be used ?
We do not evaluate expression, there are millions ways to write same bad code but by means of methods that return constant values.

**Comment by Zustin** [comment]
Hi @romani , thanks for your reply. Actually I am a static analysis fans and have a corpus to test SA tools. The reason that I submit this issue is the components of this expression are `all boolean literals` and I think checkstyle can handle this false negative based on current AST. Just expect a stronger checkstyle :)

**Comment by romani** [comment]
it would take us a lot to do internally , and it still be not complete solution. In real life such code is highly unlikely to exists.
